src/core/server/csp/csp_config.ts

```diff
@@ -58,9 +68,13 @@ export class CspConfig implements ICspConfig {
   constructor(rawCspConfig: Partial<Omit<ICspConfig, 'header'>> = {}) {
     const source = { ...DEFAULT_CONFIG, ...rawCspConfig };
 
-    this.rules = source.rules;
+    this.rules = [...source.rules];
     this.strict = source.strict;
     this.warnLegacyBrowsers = source.warnLegacyBrowsers;
-    this.header = source.rules.join('; ');
+    this.disableEmbedding = source.disableEmbedding;
+    if (!rawCspConfig.rules?.length && source.disableEmbedding) {
+      this.rules.push(FRAME_ANCESTORS_RULE);
+    }
```

Comment:
It might be a good idea to log a warning if `disableEmbedding` is enabled but custom CSP rules are used that do not include the `frame-ancestors` directive. But I think we can leave that for a future enhancement, as we don't have the ability to log when config is ingested at the moment.